ndcube/mixins/ndcubeio.py

```diff
@@ -194,43 +173,43 @@ def to_hdu(self, hdu_mask='MASK', hdu_uncertainty='UNCERT'):
 
         #------------------------------HDU1----------------------------------
         # Header for unit
-        if self.__unit:
+        if self._unit:
             header_unit = fits.Header()
-            if self.__unit is not u.dimensionless_unscaled:
-                header_unit['bunit'] = self.__unit.to_string()
-            hdus.append(fits.ImageHDU(self.__data, header_unit, name='UNIT'))
+            if self._unit is not u.dimensionless_unscaled:
+                header_unit['bunit'] = self._unit.to_string()
+            hdus.append(fits.ImageHDU(self._data, header_unit, name='DATA'))
 
         #------------------------------HDU1----------------------------------
 
         #------------------------------HDU2----------------------------------
         # Store the uncertainty
-        if self.__uncertainty is not None:
+        if self._uncertainty is not None:
             
             # Set the initial header, and the type of uncertainty
             hdr_uncertainty = fits.Header()
-            hdr_uncertainty['UTYPE'] = self.__uncertainty.uncertainty_type
+            hdr_uncertainty['UTYPE'] = self._uncertainty.uncertainty_type
 
             # Set the data of the uncertainty and header
-            hduUncert = fits.ImageHDU(self.__uncertainty.array, hdr_uncertainty, name='UTYPE')
+            hduUncert = fits.ImageHDU(self._uncertainty.array, hdr_uncertainty, name=hdu_uncertainty)
             hdus.append(hduUncert)
         #----------------------------HDU2--------------------------------
 
         #----------------------------HDU3------------------------------
         # Store the mask
-        if self.__mask is not None:
+        if self._mask is not None:
             # Always assuming that the mask is a np.ndarray
             # by checking that it has a shape
-            if not hasattr(self.__mask, 'shape'):
-                raise ValueError('only a numpy.ndarray mask can be saved.')
-            hduMask = fits.ImageHDU(self.__mask.astype(np.uint8), name=hdu_mask)
+            if not isinstance(self._mask, np.ndarray):
+                raise ValueError('Only a numpy.ndarray mask can be saved.')
+            hduMask = fits.ImageHDU(self._mask.astype(np.uint8), name=hdu_mask)
             hdus.append(hduMask)
 
         #----------------------------HDU3-------------------------------
         # Store the extra_coords
-        if self.__extra_coords is not None:
+        if self._extra_coords_wcs_axis is not None:
```

Comment:
Unlike other attributes, ```self._extra_coords_wcs_axis``` does indeed have an underscore at the start of its name.  So this is correct.